integrated_channels/blackboard/client.py

```diff
@@ -560,42 +563,104 @@ def _get_or_create_integrated_grade_column(self, bb_course_id):
-            )
-            self._delete(delete_course_content_url)
+            self.delete_course_content_from_blackboard(bb_course_id, bb_content_id)
 
+        # Remove the course
         delete_course_path = urljoin(
             self.enterprise_configuration.blackboard_base_url,
             COURSE_V3_PATH.format(course_id=bb_course_id),
         )
         resp = self._delete(delete_course_path)
         return resp
 
+    def update_integration_content_for_course(self, bb_course_id, channel_metadata_item):
+        """
+        Helper method to update the blackboard course content page with the integration custom information.
+        Differs from course customization creation in that we don't delete the course if things fail.
+        """
+        content_url = self.generate_create_course_content_url(bb_course_id)
+        error_message = ''
+        try:
+            content_resp = self._post(content_url, channel_metadata_item.get('course_content_metadata'))
+            bb_content_id = content_resp.json().get('id')
+        except ClientError as error:
+            bb_content_id = None
+            error_message = ' and received error response={}'.format(error.message)
+
+        if not bb_content_id:
+            error_message = '' if not error_message else error_message
+            raise ClientError(
+                'Something went wrong while creating course content object on Blackboard. Could not retrieve content '
+                'ID{}.'.format(error_message),
+                HTTPStatus.NOT_FOUND.value
+            )
+
+        try:
+            content_child_url = self.generate_create_course_content_child_url(bb_course_id, bb_content_id)
+            course_fully_created_response = self._post(
+                content_child_url, channel_metadata_item.get('course_child_content_metadata')
+            )
+            content_child_id = course_fully_created_response.json().get('id')
+        except ClientError as error:
+            content_child_id = None
+            error_message = ' and received error response={}'.format(error.message)
+
+        if not content_child_id:
+            error_message = '' if not error_message else error_message
+            error_handling_response = self.delete_course_content_from_blackboard(bb_course_id, bb_content_id)
+
+            raise ClientError(
+                'Something went wrong while creating course content child object on Blackboard. Could not retrieve a '
+                'content child ID and got error response={}. Deleted associated course content shell with response: '
+                '(status_code={}, body={})'.format(
+                    error_message,
+                    error_handling_response.status_code,
+                    error_handling_response.text
+                ),
+                HTTPStatus.NOT_FOUND.value
+            )
+
+        return course_fully_created_response
```

Comment:
should course_fully_created_response need to be explicitly initialized to a default value, in case we somehow fail to set in in the try block above? Figure that may not happen since _post does not raise, I think? Good to check though

Reply:
`_post` will raise if the response is >`400`. However, we catch that raise **and** if we can't retrieve an ID from that response, we raise a client error, so I don't think that a default value is needed.